examples/lua/elasticsearch_adapter.lua

```diff
@@ -0,0 +1,89 @@
+local elasticsearch = require ("elasticsearch")
+
+local client = elasticsearch.client{
+  hosts = {
+    {
+      host = "localhost",
+      port = "9200"
+    }
+  }
+}
+
+local res, status = client:index{
+  index = "rgw",
+  type = "Request",
+  id = Request.Id,
+  body = 
+	 {
+		RGWOp = Request.RGWOp,
+		DecodedURI = Request.DecodedURI,
+		ContentLength = Request.ContentLength,
+		GenericAttributes = Request.GenericAttributes,
+		Response = {
+			HTTPStatusCode = Request.Response.HTTPStatusCode,
+			HTTPStatus = Request.Response.HTTPStatus,
+			RGWCode = Request.Response.RGWCode,
+			Message = Request.Response.Message
+		},
+		SwiftAccountName = Request.SwiftAccountName,
+		Bucket = {
+			Tenant = Request.Bucket.Tenant,
+			Name = Request.Bucket.Name,
+			Marker = Request.Bucket.Marker,
+			Id = Request.Bucket.Id,
+			Count = Request.Bucket.Count,
+			Size = Request.Bucket.Size,
+			ZoneGroupId = Request.Bucket.ZoneGroupId,
+			CreationTime = Request.Bucket.CreationTime,
+			MTime = Request.Bucket.MTime,
+			Quota = {
+				MaxSize = Request.Bucket.Quota.MaxSize,
+				MaxObjects = Request.Bucket.Quota.MaxObjects,
+				Rounded = Request.Bucket.Quota.Rounded
+			},
+			PlacementRule = {
+				Name = Request.Bucket.PlacementRule.Name,
+				StorageClass = Request.Bucket.PlacementRule.StorageClass
+			},
+			User = {
+				Tenant = Request.Bucket.User.Tenant,
+				Id = Request.Bucket.User.Id
+			}
+		},
+		Object = {
+			Name = Request.Object.Name,
+			Instance = Request.Object.Instance,
+			Id = Request.Object.Id,
+			Size = Request.Object.Size,
+			MTime = Request.Object.MTime
+		},
+		ObjectOwner = {
+			DisplayName = Request.ObjectOwner.DisplayName,
+			User = Request.ObjectOwner.User
+		},
+		ZoneGroup = {
+			Name = Request.ZoneGroup.Name,
+			Endpoint = Request.ZoneGroup.Endpoint
+		},
+		BucketAcl = Request.BucketAcl,
+		ObjectAcl = Request.ObjectAcl,
+		Environment = Request.Environment,
+		RGWId = Request.RGWId,
+		HTTP = {
+			Parameters = Request.HTTP.Parameters,
+			Resources = Request.HTTP.Resources,
+			Metadata = Request.HTTP.Metadata,
```

Comment:
did you tests that assignment of non-basic data types work?
`Request.HTTP.Metadata` is a reference to a C++ map

Reply:
@pleiadesian did you try to send the metadata map? does the elasticsearch client know how to encode that?

Reply:
No, it appears as an empty record `{}`. How can I get the contents of the metadata map in Lua?

Reply:
can we do something similar to: https://github.com/ceph/ceph/blob/master/examples/lua/nats_adapter.lua#L50 ?

Reply:
Yes, it works.